core/src/main/java/fj/data/List.java

```diff
@@ -563,7 +562,57 @@ public final void foreachDoEffect(final Effect1<A> f) {
     return bind(c);
   }
 
-  /**
+    /**
+     * Traverse through the List with a might-fail function.
+     *
+     * @param f The function that might fail.
+     * @return  none if applying f fails to any element of the list or f mapped list in some .
+     */
```

Comment:
I would not describe the function f as a might fail function.  I would suggest `Traverses through the List with the given function`.  I checked the doc for Option.bind to check what it did.